test/image/make_baseline.py

```diff
@@ -0,0 +1,125 @@
+import os
+import sys
+import json
+import plotly.io as pio
+
+args = []
+if len(sys.argv) == 2 :
+    args = sys.argv[1].split()
+elif len(sys.argv) > 1 :
+    args = sys.argv
+
+root = os.getcwd()
+dirIn = os.path.join(root, 'test', 'image', 'mocks')
+dirOut = os.path.join(root, 'build', 'test_images')
+
+if '=' in args :
+    newArgs = []
+    for i in range(args.index('=') + 1, len(args)) :
+        newArgs.append(args[i])
+    args = newArgs
+    dirOut = os.path.join(root, 'test', 'image', 'baselines')
+    print('output to', dirOut)
+
+pio.templates.default = 'none'
+pio.kaleido.scope.plotlyjs = os.path.join(root, 'build', 'plotly.js')
+# TODO: specify local mathjax and plotly-geo-assets files?
+
+_credentials = open(os.path.join(root, 'build', 'credentials.json'), 'r')
+pio.kaleido.scope.mapbox_access_token = json.load(_credentials)['MAPBOX_ACCESS_TOKEN']
+_credentials.close()
+
+ALL_MOCKS = [os.path.splitext(a)[0] for a in os.listdir(dirIn) if a.endswith('.json')]
+ALL_MOCKS.sort()
+
+if len(args) > 0 :
+    allNames = [a for a in args if a in ALL_MOCKS]
+else :
+    allNames = ALL_MOCKS
+
+# gl2d pointcloud and other non-regl gl2d mock(s)
+# must be tested in certain order to work on CircleCI;
+#
+# gl-shader appears to conflict with regl.
+# We suspect that the lone gl context on CircleCI is
+# having issues with dealing with the two different
+# program binding algorithm.
+#
+# The problem will be solved by switching all our
+# WebGL-based trace types to regl.
+#
+# More info here:
+# https://github.com/plotly/plotly.js/pull/1037
+
+LAST = [
+    'gl2d_pointcloud-basic',
+    'gl2d_heatmapgl',
+    'gl2d_heatmapgl_discrete'
+]
+
+HAD = [
+    LAST[0] in allNames,
+    LAST[1] in allNames,
+    LAST[2] in allNames
+]
+
+allNames = [a for a in allNames if all(b not in a for b in LAST)]
```

Comment:
why is this not:
```py
allNames = [a for a in allNames if a not in LAST]
```
? Are the names in LAST not complete? Same Q for the blacklist below.

Reply:
Good point. Fixed in https://github.com/plotly/plotly.js/pull/5724/commits/7a44a8e603323428a998548263e9f286a5240f5a